data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaCustomConsumerFactory.java

```diff
@@ -0,0 +1,243 @@
+
+    private final StringDeserializer stringDeserializer = new StringDeserializer();
+    private String schemaType = MessageFormat.PLAINTEXT.toString();
+
+    public List<KafkaCustomConsumer> createConsumersForTopic(final KafkaConsumerConfig kafkaConsumerConfig, final TopicConfig topic,
+                                                             final Buffer<Record<Event>> buffer, final PluginMetrics pluginMetrics,
+                                                             final AcknowledgementSetManager acknowledgementSetManager,
+                                                             final AtomicBoolean shutdownInProgress) {
+        Properties authProperties = new Properties();
+        KafkaConsumerSecurityConfigurer.setAuthProperties(authProperties, kafkaConsumerConfig, LOG);
+        KafkaTopicMetrics topicMetrics = new KafkaTopicMetrics(topic.getName(), pluginMetrics);
+        Properties consumerProperties = getConsumerProperties(kafkaConsumerConfig, topic, authProperties);
+        MessageFormat schema = MessageFormat.getByMessageFormatByName(schemaType);
+
+        final List<KafkaCustomConsumer> consumers = new ArrayList<>();
+
+        try {
+            final int numWorkers = topic.getWorkers();
+            IntStream.range(0, numWorkers).forEach(index -> {
+                final KafkaConsumer kafkaConsumer;
+                switch (schema) {
+                    case JSON:
+                        kafkaConsumer = new KafkaConsumer<String, JsonNode>(consumerProperties);
+                        break;
+                    case AVRO:
+                        kafkaConsumer = new KafkaConsumer<String, GenericRecord>(consumerProperties);
+                        break;
+                    case BYTES:
+                        kafkaConsumer = new KafkaConsumer<String, byte[]>(consumerProperties);
+                        break;
+                    case PLAINTEXT:
+                    default:
+                        final GlueSchemaRegistryKafkaDeserializer glueDeserializer = KafkaConsumerSecurityConfigurer.getGlueSerializer(kafkaConsumerConfig);
+                        if (Objects.nonNull(glueDeserializer)) {
+                            kafkaConsumer = new KafkaConsumer(consumerProperties, stringDeserializer, glueDeserializer);
+                        } else {
+                            kafkaConsumer = new KafkaConsumer<String, String>(consumerProperties);
+                        }
+                        break;
+                }
+                consumers.add(new KafkaCustomConsumer(kafkaConsumer, shutdownInProgress, buffer, kafkaConsumerConfig, topic,
+                    schemaType, acknowledgementSetManager, topicMetrics));
+
+            });
+        } catch (Exception e) {
+            if (e instanceof BrokerNotAvailableException ||
+                e instanceof BrokerEndPointNotAvailableException || e instanceof TimeoutException) {
+                LOG.error("The kafka broker is not available...");
```

Comment:
Let's avoid ellipses on our errors. It seems we are unsure of something or that something else will follow. Instead, please state what Data Prepper will do for this failure case.

Also, please capitalize "Kafka" in the logs.

Here is an example (I don't know if this is actually what happens)

```
LOG.error("The Kafka broker is not available. Waiting to retry.");
```